ColorPicker.lua

```diff
@@ -339,38 +389,40 @@ for g = 1, #grpNum do
       --if it is the first cue, assign it to an executor
         if p == 1 then
         cmd('Assign Sequence '..seqCurrent..' At Executor '..execCurrent) --assign sequence to executor
-        end 
+        end
 
         --label the sequence and cues
-      cmd('Label Sequence '..seqCurrent..' \"'..grpName[g]..' color\"'); 
-      cmd('Label Sequence '..seqCurrent..' Cue '..p..' \"'..grpName[g]..' '..pName[p]..'\"');  
+      cmd('Label Sequence '..seqCurrent..' \"'..grpName[g]..' color\"');
+      cmd('Label Sequence '..seqCurrent..' Cue '..p..' \"'..grpName[g]..' '..pName[p]..'\"');
 
       --create macro and label it
-      macStore(macCurrent, grpName[g]..' '..pName[p]) 
+      macStore(macCurrent, grpName[g]..' '..pName[p])
       macLine(macCurrent, 1, 'Goto Executor '..execCurrent..' Cue '..p)
       macLine(macCurrent, 2, 'Off Macro '..macGroup.start..' Thru '..macGroup.final..' - '..macCurrent)
 
-      
+
       --change the appearance of the macro
       cmd('Appearance Macro '..macCurrent..' /color='..'"'..colSwatchBook[p]..'"')
-      
+
       --clear your programmer
-      cmd('ClearAll'); 
+      cmd('ClearAll');
 
       --calculate positions for layout pool
       posX = (p + 0.5) * (1 + spacing)
       posY = (g + 0.5) * (1 + spacing)
 
       --add macro to layout pool
-      cmd('Assign Macro '..macCurrent..' At Layout '..layoutView..'/x='..posX..' /y='..posY) 
-      
+      printNewButton(posX, posY, "Image Name", imageGrid[g][p], macCurrent, "Label")
```

Comment:
Why are the name and label hardcoded? Isn't it better to name and label all of them based on fixture and color?